core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecutorImpl.java

```diff
@@ -138,6 +145,8 @@ public RowLevelFilterExecutorImpl(List<DimColumnResolvedFilterInfo> dimColEvalua
     this.exp = exp;
     this.tableIdentifier = tableIdentifier;
     this.complexDimensionInfoMap = complexDimensionInfoMap;
+    this.dateDictionaryGenerator =
+        DirectDictionaryKeyGeneratorFactory.getDirectDictionaryGenerator(DataTypes.DATE);
```

Comment:
this issue is not happening with timestamp column?

Reply:
Yeah not happening for timestamp. But have added testcase for timestamp as well.